### PR TITLE
Unknown-fields fixes - Cleanup

### PR DESCRIPTION
**Pull request by asim-reza**
## Issue
Unknown field checks are not enabled for ProwJob configuration. https://github.com/kubernetes/test-infra/issues/26368

## Description
Prow's [checkconfig](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/checkconfig) tool is responsible for presubmit validation of ProwJob configuration. It has an optional check for unknown fields in ProwJob config that can be enabled with --warnings=unknown-fields-all. We [currently do not have this check enabled](https://github.com/kubernetes/test-infra/blob/c930a1d5f949bb9212078a05442cd39ccaae8cf3/config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml#L64-L71).
As a result it is possible to add configuration that is not recognized by Prow, causing it to be silently ignored. Common examples of this are typos in a field name and incorrect indentation.

## What was done
All errors in the configs were fixed (of which there were alot) ....

`docker run -i --rm -v "${PWD}:${PWD}" -w "${PWD}" gcr.io/k8s-prow/checkconfig:v20220905-8bcebc6376 --config-path=config/prow/config.yaml --job-config-path=config/jobs --plugin-config=config/prow/plugins.yaml --strict --warnings=mismatched-tide-lenient --warnings=tide-strict-branch --warnings=needs-ok-to-test --warnings=validate-owners --warnings=missing-trigger --warnings=validate-urls --warnings=unknown-fields --warnings=duplicate-job-refs --warnings=unknown-fields-all`

was used to perform the tests

#### Note:
 The commits were made by type of errors and there were 2 merges with the main branch as there were changes made to the repo as I was working

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml [hunk 1]
```diff
@@ -50,6 +50,8 @@ presubmits:
   # Please keep this in sync with the `ci-test-infra-prow-checkconfig` job
   - name: pull-test-infra-prow-checkconfig
```
Comment (chaodaiG) [inline config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml:51]: Please don't modify this job in place. Duplicate this job instead, aka creating a new job called `pull-test-infra-prow-checkconfig-beta` which is optional and validates with unknown-fields-all

Comment (asim-reza) [reply config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml]: done!

**Comment by asim-reza** [comment]
@chaodaiG 
> as mentioned in #27386 , please revert all changes other than the config validation job
done... I reverted all the changes and created a dublicate job

**Comment by chaodaiG** [comment]
Please squash the commits, it's weird that such a small change has 13 commits

**Comment by asim-reza** [comment]
> Please squash the commits, it's weird that such a small change has 13 commits

**Pull request closed without merging**